mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -316,6 +318,7 @@ public boolean nextKeyValue() throws IOException {
           currentCloseable.close();
           currentIterator = open(tasks.next());
         } else {
+          currentCloseable.close();
```

Comment:
It causes unclosed input stream issue if we don't close here, looks like mr framework doesn't call the final close method.

Reply:
oh,  I didn't close here. I was doing it in RecordReader#close as there are no more records to read. This is fine too.